src/main/java/com/tabnine/binary/BinaryProcessGateway.kt

```diff
@@ -20,48 +18,51 @@ open class BinaryProcessGateway {
     @Throws(IOException::class)
     open fun init(command: List<String?>?) {
         val processBuilder = ProcessBuilder(command)
+        val env = processBuilder.environment()
         if (instance.useIJProxySettings) {
-            setProxyEnvironmentVariables(processBuilder)
+            var httpConfigurable: HttpConfigurable =
+                ApplicationManager.getApplication().getComponent(HttpConfigurable::class.java)
+
+            if (httpConfigurable == null) {
+                httpConfigurable = HttpConfigurable.getInstance()
+            }
+            setProxyEnvironmentVariables(env, httpConfigurable)
+        } else {
+            setBinaryToBypassSelfHostedUrl(env)
         }
         val createdProcess = processBuilder.start()
 
         process = createdProcess
         reader = BufferedReader(InputStreamReader(createdProcess.inputStream, StandardCharsets.UTF_8))
     }
 
-    private fun setProxyEnvironmentVariables(processBuilder: ProcessBuilder) {
-        if (!HttpConfigurable.getInstance().USE_HTTP_PROXY) {
-            return
-        }
-
-        val env = processBuilder.environment()
+    private fun setBinaryToBypassSelfHostedUrl(env: MutableMap<String, String>) {
         val serverUrl = StaticConfig.getBundleServerUrl()
-        if (serverUrl.isPresent) {
-            setNoProxyForTabnineServerUrl(serverUrl, env)
-        }
-        inheritIDEProxySettings(env)
-    }
 
-    private fun setNoProxyForTabnineServerUrl(
-        serverUrl: Optional<String>,
-        env: MutableMap<String, String>
-    ) {
-        val serverUrlURL = URL(serverUrl.get())
-        val tabnineServerProxy = CommonProxy.getInstance().select(serverUrlURL).firstOrNull()
-        if (tabnineServerProxy?.type() == Proxy.Type.DIRECT) {
+        if (!serverUrl.isEmpty) {
+            val serverUrlURL = URL(serverUrl.get())
```

Comment:
maybe append to NO_PROXY env if one already defined?

Reply:
this is the case of `!useIJProxySettings` ie the user  explicitly selected to "not use the existing proxy settings"